src/lib/util/urls.ts

```diff
@@ -6,6 +8,11 @@ export function urlBasename(url: string | URL, defaultBasename = ''): string {
     return url.split('/').pop() || defaultBasename;
 }
 
-export function urlJoin(base: string | URL, sub: string): URL {
-    return new URL(sub, base);
+export function urlJoin(base: string | URL, ...subPaths: string[]): URL {
+    assert(subPaths.length >= 1);
+    let newUrl = new URL(subPaths[0], base);
+    for (const nextSub of subPaths.slice(1)) {
+        newUrl = new URL(nextSub, newUrl);
```

Comment:
```suggestion
    let newUrl = new URL(base);
    for (const subPath of subPaths) {
        newUrl = new URL(subPath, newUrl);
```
I think that's a bit more elegant 😉